packages/cli/src/cms/api/api-client.ts

```diff
@@ -13,54 +14,115 @@ import type {
   UpsertRemoteResponse,
 } from './types';
 
+interface BaseApiClient {
+  request(url: string, options: FetchWithTimeoutOptions): Promise<Response>;
+}
+type CommandOption = 'push' | 'push-status';
+export type SunsetWarning = { sunsetDate: Date; isSunsetExpired: boolean; warning: string };
+export type SunsetWarningsBuffer = SunsetWarning[];
+
 export class ReuniteApiError extends Error {
   constructor(message: string, public status: number) {
     super(message);
   }
 }
 
-class ReuniteBaseApiClient {
-  constructor(protected version: string, protected command: string) {}
-
-  protected async getParsedResponse<T>(response: Response): Promise<T> {
-    const responseBody = await response.json();
-
-    if (response.ok) {
-      return responseBody as T;
-    }
+class ReuniteApiClient implements BaseApiClient {
+  public sunsetWarnings: SunsetWarningsBuffer = [];
 
-    throw new ReuniteApiError(
-      `${responseBody.title || response.statusText || 'Unknown error'}.`,
-      response.status
-    );
-  }
+  constructor(protected version: string, protected command: string) {}
 
-  protected request(url: string, options: FetchWithTimeoutOptions) {
+  public async request(url: string, options: FetchWithTimeoutOptions) {
     const headers = {
       ...options.headers,
       'user-agent': `redocly-cli/${this.version.trim()} ${this.command}`,
     };
 
-    return fetchWithTimeout(url, {
+    const response = await fetchWithTimeout(url, {
       ...options,
       headers,
     });
+
+    this.collectSunsetWarning(response);
+
+    return response;
+  }
+
+  private collectSunsetWarning(response: Response) {
+    const sunsetTime = this.getSunsetDate(response);
+
+    if (sunsetTime) {
+      const sunsetDate = new Date(sunsetTime);
+      let sunsetWarning: SunsetWarning;
+
+      if (sunsetTime > Date.now()) {
+        sunsetWarning = {
+          sunsetDate,
+          isSunsetExpired: false,
+          warning: `Endpoint ${
+            response.url
+          } is marked for removal after ${sunsetDate.toISOString()}`,
+        };
+      } else {
+        sunsetWarning = {
+          sunsetDate,
+          isSunsetExpired: true,
+          warning: `Endpoint ${
+            response.url
+          } support expired on ${sunsetDate.toISOString()} and will be removed.`,
```

Comment:
This doesn't make sense. When the sunset date is reached, the API shouldn't be usable. Also, it's not that the person can't push, it's that they can't push from this version of Redocly CLI. Also the response.url -- what does that have to do with it? Are you migrating? or changing the version and keeping the same URL?

https://datatracker.ietf.org/doc/html/rfc8594

Reply:
>When the sunset date is reached, the API shouldn't be usable.

It's not always the case, expected sunset date might be expired but the API may still be functioning.
From the `Sunset` [header docs](https://datatracker.ietf.org/doc/html/rfc8594#section-3):

> It is safest to consider timestamps in the past mean the present
   time, meaning that the resource is expected to become unavailable at
   any time.

Reply:
>Also the response.url -- what does that have to do with it? Are you migrating? or changing the version and keeping the same URL?

`response.url` is not the `url` field from the response body, its a URL we made request to. So we just outline the URL that sent us this header.

The message mentioned above is the default (fallback) message in case we get the `Sunset` header from any endpoint, but is not the message we currently show to user. The actual message is:

https://github.com/Redocly/redocly-cli/blob/45612f32d7f7848b1e3c58dbae0c9b442276efbc/packages/cli/src/cms/api/__tests__/api.client.test.ts#L352

Reply:
I think this message should be updated. The push command is not deprecated. It's just this version of CLI can no longer work with push.

Reply:
What about 
`"push" command will be incompatible with current version of Redocly CLI after <sunsetDate>. Please run "npm install @redocly/cli@latest" to update.` 

and 

`"push" command is not supported anymore by current version of Redocly CLI. Please run "npm install @redocly/cli@latest" to update.`

?

Reply:
@RomanHotsiy @adamaltman what do you think? 👆

Reply:
The first one is okay. The second one is misleading ("not supported"). Should say:

> The "push" command is not compatible with your version of Redocly CLI. Update to the latest version by running "npm install @redocly/cli@latest"

Update the first one to:

> The "push" command will be incompatible with your version of Redocly CLI after <sunsetDate>. Update to the latest version by running "npm install @redocly/cli@latest".

Reply:
Thanks, updated!

https://github.com/Redocly/redocly-cli/blob/164c5794a27ab2d37d759bd3c9984c40c89f4971/packages/cli/src/cms/api/__tests__/api.client.test.ts#L376

https://github.com/Redocly/redocly-cli/blob/164c5794a27ab2d37d759bd3c9984c40c89f4971/packages/cli/src/cms/api/__tests__/api.client.test.ts#L401